test/integration/lidar_visual.cc

```diff
@@ -0,0 +1,572 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <gtest/gtest.h>
+
+#include <ignition/common/Console.hh>
+#include <ignition/common/Image.hh>
+#include <ignition/common/Filesystem.hh>
+
+#include "test_config.h"  // NOLINT(build/include)
+
+#include "ignition/rendering/GpuRays.hh"
+#include "ignition/rendering/LidarVisual.hh"
+#include "ignition/rendering/RenderEngine.hh"
+#include "ignition/rendering/RenderingIface.hh"
+#include "ignition/rendering/Scene.hh"
+
+#define LASER_TOL 2e-4
+#define DOUBLE_TOL 1e-6
+
+// vertical range values seem to be less accurate
+#define VERTICAL_LASER_TOL 1e-3
+
+#define WAIT_TIME 0.02
+
+using namespace ignition;
+using namespace rendering;
+
+void OnNewGpuRaysFrame(float *_scanDest, const float *_scan,
+                  unsigned int _width, unsigned int _height,
+                  unsigned int _channels,
+                  const std::string &/*_format*/)
+{
+  float f;
+  int size =  _width * _height * _channels;
+  memcpy(_scanDest, _scan, size * sizeof(f));
+}
+
+class LidarVisualTest: public testing::Test,
+                  public testing::WithParamInterface<const char *>
+{
+  // Test and verify lidar visual properties setters and getters
+  public: void Configure(const std::string &_renderEngine);
+
+  // Test boxes detection
+  public: void RaysUnitBox(const std::string &_renderEngine);
+
+  // Test vertical measurements
+  public: void LaserVertical(const std::string &_renderEngine);
+};
+
+/////////////////////////////////////////////////
+/// \brief Test LidarVisual configuraions
+void LidarVisualTest::Configure(const std::string &_renderEngine)
+{
+  if (_renderEngine == "optix")
```

Comment:
also add a check for `ogre2` for all the tests since users can still run tests using `ogre2` by doing:

```
 RENDER_ENGINE_VALUES=ogre2 make test
```